habitat_sim/simulator.py

```diff
@@ -45,13 +46,15 @@ class Configuration(object):
 
     :property sim_cfg: The configuration of the backend of the simulator
     :property agents: A list of agent configurations
-
+    :property metadata_mediator: (optional) The metadata mediator to build the simulator from
     Ties together a backend config, `sim_cfg` and a list of agent
     configurations `agents`.
     """
 
     sim_cfg: SimulatorConfiguration
     agents: List[AgentConfiguration]
+    # An existing Metadata Mediator can also be used to construct a SimulatorBackend
+    metadata_mediator: MetadataMediator = attr.ib(default=None)
```

Comment:
```suggestion
    metadata_mediator: Optional[MetadataMediator] = None
```

Reply:
Can sim_cfg and metadata_mediator override each other? Do we print a warning if that's the case? Or assert that one is None? We may want to add a validator to the effect.

Reply:
Thanks!

Reply:
> Can sim_cfg and metadata_mediator override each other? Do we print a warning if that's the case? Or assert that one is None? We may want to add a validator to the effect.

Yes, let's discuss this further in slack.